Sources/OpenTelemetry/Logging/LogHandler.swift

```diff
@@ -0,0 +1,129 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OpenTelemetry open source project
+//
+// Copyright (c) 2023 Moritz Lang and the Swift OpenTelemetry project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import InstrumentationBaggage
+import Logging
+
+public struct OTelLogHandler: LogHandler {
+    public var logLevel: Logger.Level = .info
+
+    public var metadata = Logger.Metadata()
+    public var metadataProvider: Logger.MetadataProvider?
+
+    private let label: String
+    private let processor: any OTelLogRecordProcessor
+
+    public init(
+        label: String,
+        processor: any OTelLogRecordProcessor,
+        metadataProvider: Logger.MetadataProvider? = LoggingSystem.metadataProvider
+    ) {
+        self.label = label
+        self.processor = processor
+        self.metadataProvider = metadataProvider
+    }
+
+    public subscript(metadataKey metadataKey: String) -> Logger.Metadata.Value? {
+        get {
+            metadata[metadataKey]
+        }
+        set {
+            metadata[metadataKey] = newValue
+        }
+    }
+
+    public func log(
+        level: Logger.Level,
+        message: Logger.Message,
+        metadata: Logger.Metadata?,
+        source: String,
+        file: String,
+        function: String,
+        line: UInt
+    ) {
+        var effectiveMetadata = Self.effectiveMetadata(
+            base: self.metadata,
+            provider: metadataProvider,
+            explicit: metadata
+        )
+
+        let traceIDBytes: [UInt8]?
+        let spanIDBytes: [UInt8]?
+
+        if let traceIDMetadata = effectiveMetadata?["trace_id"],
```

Comment:
@ktoso This is what I was talking about regarding trace & span IDs being fields. The approach I took with this PR is to set `.stringConvertible` metadata inside the OTel logging metadata provider instead of plain strings so that I'm able to cast them back into their original types here.

Alternatively, instead of reading trace/span IDs from `effectiveMetadata` I could also look up `Baggage.current?.spanContext` here.

Reply:
Hm, I see, that's quite tricky... Does the metadata provider work _well_ in other log handlers then if you'd move it to this custom string convertible trick? I think either is fine probably here

Reply:
> Does the metadata provider work _well_ in other log handlers then if you'd move it to this custom string convertible trick?
> @ktoso 

I'd think so. What else are you going to do with a custom string convertible of which you don't know the underlying than to convert it into a `String`?

Reply:
Yeah the only thing others can do is convert it to a string, so the `description` must provide a format that works well in metadata and some e.g. some `MyType(bla: bla)` :) Just checking it'll work for others, if it does then this is okey I think

Reply:
The description is the textual representation of the IDs, so no Swift struct like looking thing but a hex-representation of the ID bytes, the same that you see in the `traceparent` header. That way, when using OTel for tracing but a different handler for logging the backend should still be able to correlate the two.

Reply:
That sounds like it'll work fine then :)